app/controllers/sign_up_sheet_controller.rb

```diff
@@ -208,6 +208,40 @@ def list
     end
   end
 
+  def sign_up
+    # find the assignment to which user is signing up
+    @assignment = AssignmentParticipant.find(params[:id]).assignment
+    @user_id = session[:user].id
+    # Always use team_id ACS
+    # s = Signupsheet.new
+    # Team lazy initialization: check whether the user already has a team for this assignment
+    unless SignUpSheet.signup_team(@assignment.id, @user_id, params[:topic_id])
+      flash[:error] = "You've already signed up for a topic!"
+    end
+    redirect_to action: 'list', id: params[:id]
+  end
+
+  # routes to new page to specficy student
+  def signup_as_instructor; end
+
+  def signup_as_instructor_action
```

Comment:
There might be a better name for this method.
The ABC complexity is a little bit high.